prediction/src/algorithms/training/classifier/trainval_detector.py

```diff
@@ -0,0 +1,160 @@
+        coord = Variable(coord.cuda(async=True), volatile=True)
+
+        _, output = net(data, coord)
+        loss_output = loss(output, target, train=False)
+
+        loss_output[0] = loss_output[0].data[0]
+        metrics.append(loss_output)
+    end_time = time.time()
+
+    metrics = np.asarray(metrics, np.float32)
+    print('Validation: tpr %3.2f, tnr %3.8f, total pos %d, total neg %d, time %3.2f' % (
+        100.0 * np.sum(metrics[:, 6]) / np.sum(metrics[:, 7]),
+        100.0 * np.sum(metrics[:, 8]) / np.sum(metrics[:, 9]),
+        np.sum(metrics[:, 7]),
+        np.sum(metrics[:, 9]),
+        end_time - start_time))
+    print('loss %2.4f, classify loss %2.4f, regress loss %2.4f, %2.4f, %2.4f, %2.4f' % (
+        np.mean(metrics[:, 0]),
+        np.mean(metrics[:, 1]),
+        np.mean(metrics[:, 2]),
+        np.mean(metrics[:, 3]),
+        np.mean(metrics[:, 4]),
+        np.mean(metrics[:, 5])))
+    print
+    print
+
+
+def test_nodulenet(data_loader, net, get_pbb, save_dir, config, n_per_run):
+    start_time = time.time()
+    save_dir = os.path.join(save_dir, 'bbox')
+    if not os.path.exists(save_dir):
+        os.makedirs(save_dir)
+    net.eval()
+    namelist = []
+    split_comber = data_loader.dataset.split_comber
+    for i_name, (data, target, coord, nzhw) in enumerate(data_loader):
+        s = time.time()
+        target = [np.asarray(t, np.float32) for t in target]
+        lbb = target[0]
+        nzhw = nzhw[0]
+        name = data_loader.dataset.filenames[i_name].split('-')[0].split('/')[-1]
+        data = data[0][0]
+        coord = coord[0][0]
+        isfeat = False
+        if 'output_feature' in config:
+            if config['output_feature']:
+                isfeat = True
+        print(data.size())
+        splitlist = range(0, len(data) + 1, n_per_run)
+        if splitlist[-1] != len(data):
+            splitlist.append(len(data))
+        outputlist = []
+        featurelist = []
+
+        for i in range(len(splitlist) - 1):
+            input = Variable(data[splitlist[i]:splitlist[i + 1]], volatile=True).cuda()
+            inputcoord = Variable(coord[splitlist[i]:splitlist[i + 1]], volatile=True).cuda()
+            _, output = net(input, inputcoord)
+            outputlist.append(output.data.cpu().numpy())
+        output = np.concatenate(outputlist, 0)
+        output = split_comber.combine(output, nzhw=nzhw)
+        thresh = -3
+        pbb, mask = get_pbb(output, thresh, ismask=True)
+        #tp,fp,fn,_ = acc(pbb,lbb,0,0.1,0.1)
+        # print([len(tp),len(fp),len(fn)])
+        print([i_name, name])
+        e = time.time()
+        np.save(os.path.join(save_dir, name + '_pbb.npy'), pbb)
+        np.save(os.path.join(save_dir, name + '_lbb.npy'), lbb)
+    np.save(os.path.join(save_dir, 'namelist.npy'), namelist)
+    end_time = time.time()
+
+    print('elapsed time is %3.2f seconds' % (end_time - start_time))
+    print
```

Comment:
I think you can achieve the same by appending two '\n' to the previous `print` statement :)

Reply:
also, that last `print` statement will error in py3 since `print` is a function.